### ISSUE TITLE
Fix crash and re-enable background sync for Android 10 

### ISSUE THREAD

**Issue by semanticer** [toggl/mobileapp#6147]
## 💥 Crash Report
With [this merge](https://github.com/toggl/mobileapp/pull/6059), we disabled background sync on Android 10 because we encountered [this crash](https://github.com/toggl/mobileapp/issues/6040). We should fix the crash and re-enable sync for Android 10. 

### 👥 Impact
This affected around 0.08% of our users. 
But only users with OG Pixel and Pixel 3a with Android 10  

### 🛰 Link
https://appcenter.ms/orgs/Toggl/apps/Toggl-Android/crashes/errors/1263678719u/overview

### 📝  Exception & Stack Trace
`com.microsoft.appcenter.crashes.Crashes.saveUncaughtException`
```cs
java.lang.RuntimeException: An error occurred while executing doInBackground()
  at android.os.AsyncTask$4.done(AsyncTask.java:399)
  at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
  at java.util.concurrent.FutureTask.run(FutureTask.java:271)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
  at java.lang.Thread.run(Thread.java:919)
  Caused by: java.lang.SecurityException: Caller no longer running, last stopped +3s450ms because: timed out while starting
    at android.os.Parcel.createException(Parcel.java:2071)
    at android.os.Parcel.readException(Parcel.java:2039)
    at android.os.Parcel.readException(Parcel.java:1987)
    at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:292)
    at android.app.job.JobParameters.dequeueWork(JobParameters.java:248)
    at android.support.v4.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
    at android.support.v4.app.JobIntentService.dequeueWork(JobIntentService.java:640)
    at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
    at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
    at android.os.AsyncTask$3.call(AsyncTask.java:378)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:919)
    Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:493)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:371)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:169)
	at android.os.Binder.execTransactInternal(Binder.java:1021)
```

**Comment by semanticer** [comment]
We can try to launch a foreground service.

**Comment by daividssilverio** [comment]
It turns out we've disable the wrong `Service` in #6059 and the crash was happening in one of our other `JobIntentServices`.

Why?
- `BackgroundSyncJobSchedulerService` is a `JobService`;
- The crash reported refers to a `JobIntentService`.

2.9 should include a fix for that crash that was added by #6839 .

I think we can simply remove the changes made on #6059.

**Comment by semanticer** [comment]
Looks like you are right. I will prepare PR for reverting the changes back.
